tests/functional/RemoteWebElementTest.php

```diff
@@ -117,6 +118,65 @@ public function testShouldClick()
         );
     }
 
+    /**
+     * Test that workarounds to geckodriver#653 works as expected.
+     *
+     * @covers ::click
+     * @covers ::clickChildElement
+     * @dataProvider provideValidClick
+     * @param string $id
+     */
+    public function testShouldClickOnBlockLevelElement($id)
```

Comment:
I know its generally bad idea in tests, but considering the time it takes to initialize new browser, I think the scales are a bit different with independency for integration tests of this kind - so would it be an issue by butting all the four cases into single test method?